pkg/generators/extension.go

```diff
@@ -27,18 +27,33 @@ import (
 
 const extensionPrefix = "x-kubernetes-"
 
-// Extension tag to openapi extension
-var tagToExtension = map[string]string{
-	"patchMergeKey": "x-kubernetes-patch-merge-key",
-	"patchStrategy": "x-kubernetes-patch-strategy",
-	"listType":      "x-kubernetes-list-type",
-	"listMapKey":    "x-kubernetes-list-map-keys",
+// extensionAttributes encapsulates common traits for particular extensions.
+type extensionAttributes struct {
+	xName         string
+	kind          types.Kind
+	allowedValues sets.String
 }
 
-// Enum values per extension
-var allowedExtensionValues = map[string]sets.String{
-	"x-kubernetes-patch-strategy": sets.NewString("merge", "retainKeys"),
-	"x-kubernetes-list-type":      sets.NewString("atomic", "set", "map"),
+// Extension tag to openapi extension attributes
+var tagToExtension = map[string]extensionAttributes{
```

Comment:
The number of conditions has increased in that pull-request. I think this PR goes back to what we were initially discussing: we should have some sort of virtual dispatch with `extensionAttributes` being an interface, with different types of attributes. 

For example, things that are not looking as targeted as they could be:
- `allowedValues` is optional rather than mandatory, creating a new corner-case
- it's not clear if `kind` is mandatory or not, but that creates a new corner-case
- I'm expecting a few more corner cases when you'll want to write more specific `kind` validation
- you still have a list of values in `extension` where we might not have a list (a bunch of new corner-cases)
- you have a list of strings in `extension`, even though they might not always be

For now it's fine (though it's less and less), you can keep going in that direction, but my opinion is that it's going to become more and more obvious that many of these things should be handled by a virtual dispatch.